azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java

```diff
@@ -81,22 +83,39 @@ ServiceClientCredentials getCredentials() {
             final String msiEndpoint = getProperty(this.environment, "MSI_ENDPOINT");
             final String msiSecret = getProperty(this.environment, "MSI_SECRET");
             return new AppServiceMSICredentials(AzureEnvironment.AZURE, msiEndpoint, msiSecret);
-        } else if (this.environment.containsProperty(Constants.AZURE_CLIENTID)
-                && this.environment.containsProperty(Constants.AZURE_CLIENTKEY)) {
+        }
+
+        final long timeAcquiringTimeoutInSeconds = this.environment.getProperty(
+                Constants.AZURE_TOKEN_ACQUIRE_TIMEOUT_IN_SECONDS, Long.class, Constants.TOKEN_ACQUIRE_TIMEOUT_SECS);
+        if (this.environment.containsProperty(Constants.AZURE_KEYVAULT_CLIENT_ID)
+                && this.environment.containsProperty(Constants.AZURE_KEYVAULT_CLIENT_KEY)) {
             LOG.debug("Will use custom credentials");
-            final long timeAcquiringTimeoutInSeconds = this.environment.getProperty(
-                    Constants.AZURE_TOKEN_ACQUIRE_TIMEOUT_IN_SECONDS, Long.class, Constants.TOKEN_ACQUIRE_TIMEOUT_SECS);
-            final String clientId = getProperty(this.environment, Constants.AZURE_CLIENTID);
-            final String clientKey = getProperty(this.environment, Constants.AZURE_CLIENTKEY);
+            final String clientId = getProperty(this.environment, Constants.AZURE_KEYVAULT_CLIENT_ID);
+            final String clientKey = getProperty(this.environment, Constants.AZURE_KEYVAULT_CLIENT_KEY);
             return new AzureKeyVaultCredential(clientId, clientKey, timeAcquiringTimeoutInSeconds);
-        } else if (this.environment.containsProperty(Constants.AZURE_CLIENTID)) {
+        }
+
+        if (this.environment.containsProperty(Constants.AZURE_KEYVAULT_CLIENT_ID) &&
```

Comment:
It would be best to refactor the fallback process when getting credential. Too much details in just one method IMO.
Leaving the decision to you.

Reply:
Will do this in another PR.